QModManager/Utility/IOUtilities.cs

```diff
@@ -0,0 +1,180 @@
+                    else 
+                        toWrite += $"{GenerateSpaces(0)}`---- {fileinfo.Name} ({ParseSize(fileinfo.Length)})\n";
+                }
+
+                return toWrite;
+            }
+            catch (Exception e)
+            {
+                throw e;
+            }
+        }
+        private static string GetFolderStructureRecursively(string directory, int spaces = 0)
+        {
+            try
+            {
+                DirectoryInfo dirInfo = new DirectoryInfo(directory);
+                string toWrite = $"{GenerateSpaces(spaces)}|---+ {dirInfo.Name} ({ParseSize(CalculateFolderSizeRecursively(directory))})\n";
+
+                if (BannedFolders.Contains(dirInfo.Name) || BannedFolders.Contains($"{dirInfo.Parent.Name}/{dirInfo.Name}"))
+                {
+                    toWrite += $"{GenerateSpaces(spaces + 4)}`---- ({GetFileCountRecursively(directory)} elements not shown...)\n";
+                    return toWrite;
+                }
+
+                foreach (string dir in Directory.GetDirectories(directory))
+                {
+                    toWrite += GetFolderStructureRecursively(dir, spaces + 4);
+                }
+
+                string[] files = Directory.GetFiles(directory);
+                for (int i = 1; i <= files.Length; i++)
+                {
+                    FileInfo fileinfo = new FileInfo(files[i - 1]);
+                    if (i != files.Length)
+                        toWrite += $"{GenerateSpaces(spaces + 4)}|---- {fileinfo.Name} ({ParseSize(fileinfo.Length)})\n";
+                    else
+                        toWrite += $"{GenerateSpaces(spaces + 4)}`---- {fileinfo.Name} ({ParseSize(fileinfo.Length)})\n";
+                }
+
+                return toWrite;
+            }
+            catch (Exception e)
+            {
+                throw e;
+            }
+        }
+
+        private static string ParseSize(long lsize)
+        {
+            float size = lsize;
+            string unit;
+
+            if (size < 1024)
+            {
+                unit = "B";
+            }
+            else
+            {
+                size /= 1024;
+                if (size < 1024)
+                {
+                    unit = "KB";
+                }
+                else
+                {
+                    size /= 1024;
+                    if (size < 1024)
+                    {
+                        unit = "MB";
+                    }
+                    else
+                    {
+                        size /= 1024;
+                        if (size < 1024)
+                        {
+                            unit = "GB";
+                        }
+                        else
+                        {
+                            size /= 1024;
+                            unit = "TB";
+                        }
+                    }
+                }
+            }
+
+            string number = size.ToTwoDecimalString();
```

Comment:
string.Format("F2");  
https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings

Reply:
This extension method is from `Assembly-CSharp-firstpass`, I don't think there's any harm in using it.

Reply:
After looking at the method in dnSpy, this is exactly what it does. Changed in commit 1a86b08

![dnSpy screenshot of ToTwoDecimalString](https://user-images.githubusercontent.com/32238504/53694457-03e31000-3db8-11e9-9446-01c2a5601cdf.png)